lib/veres-one-client.js

```diff
@@ -0,0 +1,187 @@
+/*!
+ * Copyright (c) 2018 Veres One Project. All rights reserved.
+ */
+'use strict';
+
+const axios = require('axios');
+const https = require('https');
+const WebLedgerClient = require('web-ledger-client');
+
+const DEFAULT_MODE = 'test';
+
+class VeresOneClient {
+  /**
+   * @param injector {Injector}
+   *
+   * @param hostname {string} Hostname of the ledger (points to a load balancer
+   *   or a specific node).
+   * @param ledger {WebLedgerClient}
+
+   * @param [mode='test'] {string} One of 'dev'/'test'/'live'. Determines https
+   *   agent settings (does not reject unsigned certs in 'dev' mode, for
+   *   example).
+   * @param [logger]
+   * @param [httpsAgent] {Agent}
+   */
+  constructor({injector, hostname, ledger, mode, logger, httpsAgent}) {
+    this.injector = injector;
+    this.hostname = hostname;
+    if(!hostname) {
+      throw new Error('Missing ledger hostname');
+    }
+    this.ledger = ledger ||
+      new WebLedgerClient({injector, hostname, mode, logger, httpsAgent});
+
+    this.mode = mode || DEFAULT_MODE;
+    this.logger = logger || console;
+
+    this.httpsAgent = httpsAgent;
+    if(mode === 'dev' && !this.httpsAgent) {
+      this.httpsAgent = new https.Agent({rejectUnauthorized: false});
+    }
+  }
+
+  /**
+   * Fetches a DID Document for a given DID. If it contains a #hash fragment,
+   * it's likely a key id, so just return the subgraph, not the full doc.
+   *
+   * @param did {string} DID uri (possibly with hash fragment)
+   *
+   * @returns {Promise<object>} Resolves to DID Document Fetch Result
+   */
+  async get({did}) {
+    if(!did) {
+      throw new Error('Invalid or missing DID URI');
```

Comment:
```js
if(!(did && typeof did === 'string')) {
  throw new TypeError('"did" must be a string representing a DID URI.');
}
```

Reply:
Why check for it being a string? The main thing this is guarding against is a missing DID, not a non-string one.